### PR TITLE
Batch SDK update

### PR DESCRIPTION
**Pull request by annatisch**
Updated tests to use new test SDK (@troydai)
Updated to latest Batch SDK (breaking change)

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### CODE REVIEW

**Comment by troydai** [comment]
@annatisch can you take a look the style failures on CI? You can run `run_styles --ci` to verify formatting locally.

**Comment by annatisch** [comment]
@troydai  - yup I'm doing that now! Sorry totally forgot to run the style check...

**Comment by troydai** [comment]
@annatisch I'm reviewing the pull request now. Thanks for fixing the styles.

File: src/command_modules/azure-cli-batch/tests/batch_preparers.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# --------------------------------------------------------------------------------------------
+
+import os
+from azure.cli.testsdk import CliTestError, ResourceGroupPreparer
+from azure.cli.testsdk.preparers import AbstractPreparer, SingleValueReplacer
+from azure.cli.testsdk.base import execute
+
+
+class BatchAccountPreparer(AbstractPreparer, SingleValueReplacer):
+    def __init__(self, name_prefix='clibatch',  # pylint: disable=too-many-arguments
```
Comment (troydai) [inline src/command_modules/azure-cli-batch/tests/batch_preparers.py]: The `too-many-arguments` rule is disabled globally now. So you can remove the disable statement.

**Comment by troydai** [comment]
Looks good to me. This can be merged and we will try run this live in following days.

**Comment by troydai** [comment]
I'll wait for @xingwu1 to take a further look.

**Comment by annatisch** [comment]
Thanks @troydai!
@xingwu1 - would you be able to take a quick look?

File: src/command_modules/azure-cli-batch/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,13 @@
 Release History
 ===============
 
+3.0.0 (2017-05-29)
+++++++++++++++++++
+
+* Updated to Batch SDK 3.0.0 with support for low-priority VMs in pools.
```
Comment (xingwu1) [inline src/command_modules/azure-cli-batch/HISTORY.rst]: I know it is not exposed to command line. However, do we need to mention more update? for example, outfiles.

**Pull request merged**